enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,476 @@
+   cookie=0xdeff105, duration=48397.416s, table=5, n_packets=5, n_bytes=1974, ip actions=ct(commit,table=2,zone=64001,nat)
+   cookie=0xdeff105, duration=10.170s, table=2, n_packets=16588, n_bytes=1428836, actions=mod_dl_dst:9e:0c:05:73:23:54,output:"patch-br-ex_rhe"
+   ```
+4. The loadbalancer in OVN GR will then handle unSNAT, unDNAT to src_ip
+   (10.43.117.48), dest_ip(169.254.169.2) and send the packet back to br-ex.
+
+5. br-ex will unDNAT and send the packet to localhost.
+   
+   ```none
+   cookie=0xdeff105, duration=13742.995s, table=0, n_packets=2198, n_bytes=306654, priority=500,ip,in_port="patch-br-ex_rhe",nw_src=10.43.0.0/16,nw_dst=169.254.169.2 actions=ct(table=3,zone=64001,nat)
+   cookie=0xdeff105, duration=14.962s, table=3, n_packets=2198, n_bytes=306654, actions=move:NXM_OF_ETH_DST[]->NXM_OF_ETH_SRC[],mod_dl_dst:16:38:a3:9b:fd:6b,LOCAL
+   ```
+   
+### Host to NodePort Services Backed by Host Pods
+
+This traffic case is almost the same as the previous one. The only difference is
+that before the packet is sent to br-ex, it will be DNATed to the service's
+clusterIP by host iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+### External to NodePort Services
+
+The general flow is:
+
+1. The packet enters host via eth0.
+2. In the host, the packet is DNATed to the service's clusterIP by the following iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+3. The packet is forwarded to br-ex bridge.
+
+The rest of the steps is the same as the normal local gateway mode.
+
+### Workflow Description
+
+#### Deploying
+
+There will be no change to the MicroShift deployment procedure.
+
+#### Upgrading
+
+The microshift-ovs-init.service will be responsible for the OVS bridge change.
+As we will use a dummy interface for br-ex bridge, the `gatewayInterface` flag
+in `/etc/microshift/ovn.yaml` will be deprecated. Also, as the configuration of
+OVS will not affect host physical interfaces, there is no need to allow user to
+config the OVS bridges. The flag `disableOVSInit` can also be deprecated.
+
+#### Configuring
+
+There is no need to introduce any new configuration flag for this change. Unless
+we want to allow user to specify the IP address used by the br-ex interface.
+
+#### Deploying Applications
+
+Applications are deployed as usual.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+On the OVN-Kubernetes side, we need to change the local gateway mode:
+
+1. Allow using an external gateway bridge without uplink port
+2. Add static routes for all host network with the ovn-k internal masquerade
+   next-hop.
```

Comment:
Can you clarify in which cases this will be done as well?

Reply:
`for all host networks` means all the local IPs and direct routes of the node. In our case, they're `10.44.0.0/32` and `10.89.0.0/24`.
I think this will be part of @trozet 's fix for the no default gateway scenario. So this will be the default for all cases in OVN-K.

Reply:
> `for all host networks` means all the local IPs and direct routes of the node. In our case, they're `10.44.0.0/32` and `10.89.0.0/24`.

What about other routes the node might have? As I asked before, would we monitor the host routing table for modifications?

Reply:
This will be coved by the fix of @trozet. Let's keep this discussion in the thread above.

Reply:
This will be coved by the fix of @trozet. Let's keep this discussion in the thread above.

Reply:
OK, but I think `all host network` is not clear. I would rephrase to be more specific.